dll/win32/shell32/wine/shellpath.c

```diff
@@ -1866,7 +1866,7 @@ INT Shell_ParseSpecialFolder(_In_ LPCWSTR pszStart, _Out_ LPWSTR *ppch, _Out_ IN
     {
         *ppch = (LPWSTR)(pchBackslash + 1);
         *pcch = (pchBackslash - pszStart) + 1;
-        StrCpyNW(szPath, pszStart, max(*pcch, _countof(szPath)));
+        StrCpyNW(szPath, pszStart, min(*pcch, _countof(szPath)));
```

Comment:
nice bugfix.

Now, I wonder what would be the better course of action, if the szPath is not large enough to contain the string pointed by pszStart (and going up to the backslash)...

Reply:
Using `CStringW` may be a solution.

Reply:
No. The special folder ID (for `SHGetSpecialFolderID`) shouldn't be beyond `MAX_PATH`.